frontend/server/src/main/java/org/pytorch/serve/metrics/format/prometheous/PrometheusMetric.java

```diff
@@ -0,0 +1,31 @@
+package org.pytorch.serve.metrics.format.prometheous;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.pytorch.serve.metrics.IMetric;
+import org.pytorch.serve.metrics.MetricBuilder;
+
+public abstract class PrometheusMetric extends IMetric {
+    public PrometheusMetric(
+            MetricBuilder.MetricContext context,
+            MetricBuilder.MetricType type,
+            String name,
+            String unit,
+            List<String> dimensionNames) {
+        super(context, type, name, unit, dimensionNames);
+        // Hostname dimension is automatically included for backend metrics
+        if (this.context == MetricBuilder.MetricContext.BACKEND) {
+            this.dimensionNames.add("Hostname");
+        }
+    }
```

Comment:
it is not necessary to check for each backend metric. the metric parser can do it once. eg. if "model_metrics" then directly add hostname in dimensions